src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11066,7 +11066,9 @@ namespace UnitarySystems {
 
             } else if ((SELECT_CASE_var == DataHVACGlobals::Coil_HeatingGasOrOtherFuel) ||
                        (SELECT_CASE_var == DataHVACGlobals::Coil_HeatingElectric)) {
-                HeatCoilLoad = PartLoadRatio * m_DesignHeatingCapacity;
+                if ((HeatCoilLoad != DataLoopNode::SensedLoadFlagValue) || (PartLoadRatio > 0)) {
+                        HeatCoilLoad = PartLoadRatio * m_DesignHeatingCapacity;
+                }
```

Comment:
OK, now I understand what is happening, the question is whether this is the correct fix or not.

The heating coil is controlled 2 ways, either by load or by set point.

        // Control output to meet load QCoilReq (QCoilReq is passed in if load controlled, otherwise QCoilReq=-999)
        if ((AirMassFlow > 0.0 && HeatingCoil(CoilNum).NominalCapacity > 0.0) && (GetCurrentScheduleValue(HeatingCoil(CoilNum).SchedPtr) > 0.0) &&
            (QCoilReq > 0.0)) {

or

            // Control coil output to meet a setpoint temperature.
        } else if ((AirMassFlow > 0.0 && HeatingCoil(CoilNum).NominalCapacity > 0.0) &&
                   (GetCurrentScheduleValue(HeatingCoil(CoilNum).SchedPtr) > 0.0) && (QCoilReq == SensedLoadFlagValue) &&
                   (std::abs(TempSetPoint - TempAirIn) > TempControlTol)) {

The code here in UnitarySystem is passing a load to the heating coil based on PLR * DesignHeatingCapacity. This seems correct to me.

I would question why the PLR is being set to 0 around UnitarySystem line 12850 in the else if since the previous call to the heating coil should have return the results with the heating coil off.

                //     IF outlet temp at no load is within ACC of set point, do not run the coil
                if (std::abs(DataLoopNode::Node(OutletNode).Temp - DesOutTemp) < Acc ||
                    this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_UserDefined) {
                    // do nothing, coil is at the set point.
                } else if ((DataLoopNode::Node(OutletNode).Temp - DesOutTemp) > Acc) { // IF outlet temp is above set point turn off coil
                    PartLoadFrac = 0.0;
                } else { // ELSE get full load result
 
Up at line 12783 (ish) is this:

                    } else if ((SELECT_CASE_var == DataHVACGlobals::Coil_HeatingGasOrOtherFuel) ||
                               (SELECT_CASE_var == DataHVACGlobals::Coil_HeatingElectric) ||
                               (SELECT_CASE_var == DataHVACGlobals::Coil_HeatingDesuperheater)) {

                        HeatingCoils::SimulateHeatingCoilComponents(
                            state, CompName, FirstHVACIteration, DataLoopNode::SensedLoadFlagValue, CompIndex, _, _, FanOpMode);

It seems to me that this call is wrong and should be:

                        HeatingCoils::SimulateHeatingCoilComponents(
                            state, CompName, FirstHVACIteration, PartLoadFrac, CompIndex, _, _, FanOpMode);

or whatever it needs to be such that the coil is off. Also there are 3 coil types here, so be careful since one coil type may require a different form of the call. If that's the case then this else if should be split into 2. I would test all 3 coil types with whatever change is made here. I assume once that is done then this change (line 11069 above) can be reverted.